system/mount.py

```diff
@@ -378,7 +435,7 @@ def main():
         module.exit_json(changed=changed, **args)
 
     if state == 'unmounted':
-        if ismount(name):
+        if ismount(name) or is_bind_mounted(module, name):
```

Comment:
(minor non-blocking comment on pre-existing code) The ordering of the 'if check_mode' and 'if ismount' statements is different between the 'absent' stanza, and the 'unmounted' and 'mounted' stanzas. That may be necessary for the logic, but it looks a little odd reading through the code.